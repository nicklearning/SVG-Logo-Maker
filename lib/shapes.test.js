const { Circle, Triangle, Square } = require('./shapes');

describe('Shapes', () => {
    describe('circle', () => {
        it('should construct a circle with the correct values', () => {
            const shape = new Circle("green", "white", "NJS");
            expect(shape.render()).toMatchSnapshot(`<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
                <circle cx="150" cy="100" r="80" fill="${shape.shapeColor}"></circle>
                <text x="150" y="122.5" font-size="60" text-anchor="middle" fill="${shape.textColor}">${shape.text}</text>
          </svg>`)
        });
    });
    describe('triangle', () => {
        it('should construct a triangle with the correct values', () => {
            const shape = new Triangle("green", "white", "NJS");
            expect(shape.render()).toMatchSnapshot(`<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
            <polygon points="${shape.x1},${shape.y1} ${shape.x2},${shape.y2} ${shape.x3},${shape.y3}" fill="${shape.shapeColor}"></polygon>
            <text x="180" y="175" font-size="60" text-anchor="middle" fill="${shape.textColor}">${shape.text}</text>
          </svg>`)
        });
    });
    describe('square', () => {
        it('should construct a square with the correct values', () => {
            const shape = new Square("green", "white", "NJS");
            expect(shape.render()).toMatchSnapshot(`<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
            <rect x='80' y='20' width="${shape.sideLength}" height="${shape.sideLength}" fill="${shape.shapeColor}"></rect>
            <text x="160" y="120" font-size="60" text-anchor="middle" fill="${shape.textColor}">${shape.text}</text>
          </svg>`)
        });
    });
});