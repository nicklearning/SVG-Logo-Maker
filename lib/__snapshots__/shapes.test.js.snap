// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapes circle should construct a circle with the correct values: <svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
                <circle cx="150" cy="100" r="80" fill="green"></circle>
                <text x="150" y="122.5" font-size="60" text-anchor="middle" fill="white">NJS</text>
          </svg> 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
        <circle cx="150" cy="100" r="80" fill="green"></circle>
        <text x="150" y="122.5" font-size="60" text-anchor="middle" fill="white">NJS</text>
      </svg>"
`;

exports[`Shapes square should construct a square with the correct values: <svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
            <rect x='80' y='20' width="160" height="160" fill="green"></rect>
            <text x="160" y="120" font-size="60" text-anchor="middle" fill="white">NJS</text>
          </svg> 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
              <rect x='80' y='20' width="160" height="160" fill="green"></rect>
              <text x="160" y="120" font-size="60" text-anchor="middle" fill="white">NJS</text>
            </svg>"
`;

exports[`Shapes triangle should construct a triangle with the correct values: <svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
            <polygon points="undefined,undefined undefined,undefined undefined,undefined" fill="green"></polygon>
            <text x="180" y="175" font-size="60" text-anchor="middle" fill="white">NJS</text>
          </svg> 1`] = `
"<svg xmlns="http://www.w3.org/2000/svg" width="300" height="200">
              <polygon points="100,200 260,200 180,20" fill="green"></polygon>
              <text x="180" y="175" font-size="60" text-anchor="middle" fill="white">NJS</text>
            </svg>"
`;
